### PR TITLE
Restore default border and focus style on image url input field.

### LINKED ISSUE #58494: Image: URL input field insufficient focus style
### Description

The image link URL implementation has always been slighly inconsistent with the normal Link UI, which is not ideal, but at least provided some level of accessibility.

Starting from WordPress 6.4, the focus style of the input field has been changed and it is now less visible and also inconsistent with the default input fields focus style. Making the focus style _less_ visible isn't an improvement. Rather, it's a regression.

Specifically, the CSS sets the input border width to `0` only for visual aesthetics reasons but that impacts the focus style. IN fact the focus style relies on the border being visible. It appears this change hasn'g ben tested with keyboard and accessibility wasn't taken into account.

Screenshot from WordPress 6.3:

The focus style and overall styling of the input field is inconsistent with the default input styling but at least the focus style is a thick blue visual outline:

Screenshot from WordPress 6.4:

The focus style is thinner and barely visible.

Screenshot from WordPress trunk and Gutenberg trunk:

Same on trunk. I'd also say the horizontal alignment of the input and placeholder text isn't nice as it used to be.

trunk

Cc @annezazu can we please make sure this gets fixed for WP 6.5? Thanks 🙏 

### Step-by-step reproduction instructions

- Add an image block.
- In the block toolbar, click the Link button.
- Initial focus is set to the input field.
- Observe the focus style is worse than in previous versions.
- Observe the horizontal alignment of the placeholder text could be improved.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by afercia**
Fixes https://github.com/WordPress/gutenberg/issues/58494

## What?

The UI to inseert link on images is different from the default URL input UI and has now an insufficient, and inconsistent, focus style. This UI is used in a few blocks e.g. Image, Media & Text, etc.

## Why?

Focus style must always be clear. Ideally it should be consistent across all the input fields in the editor UI.
There's no good reason why the image URL input mut look and work differently from the default URL input UI used for text, besides very specific features that are only available for images.

This PR aims to make the image URL input use the default focus style. To do so, some additional CSS changes are necessary like adjusting the input and button sizes to the new defaults and better spacing and alignment. I tried to keep the changest o a minimum. 

There's more inconsistencies to fix, both visual and functional. I'll create a follow=up issue.

## How?

- Adjusts the CSS to use the default focus style for input fields.
- Restores the input border: the border is integrant part of the focus style and should never be overridden with 'ad hoc' CSS.
- Adjust input and button sizes.
- Adjusts spacing around the input and alignments.

### Testing Instructions for Keyboard

- HInt: compare with the 'default' URL input UI used for links on text content.
- Add a link to an image for example in a Media & Text block or an Image block.
- The image URL input popover opens.
- Initial focus is set on the input field.
- Observe the focus style is the default one used for all input fields.
- Observe the input field and buttons sizes is nor bigger.
- Observe spacing and alignments are closer to the ones of the default URL input UI.

## Screenshots or screencast 

Before: non focused and focused:

before 01

After: non focused and focused:

after 01

Compare with the default URL input UI:

default 01

### CODE REVIEW

**Comment by afercia** [comment]
As I mentioned in this PR description, there's more inconsistencies to fix:
- The button placement: visually placed inside or outside the input field.
- The aria-disabled state of the Submit button.
- The labelling of the button: Apply vs. Submit.
- Different placeholder text.
- Divider border, spacing, and other CSS details.

I will create a follow-up issue to keep changes in this PR to a minimum.

**Review by carolinan — APPROVED** [review]
I have tested the change with the media & text block and image block.
The focus style is corrected and the size and alignments of the button matches the one in the link control that is used for the paragraph.

**Comment by afercia** [comment]
@WordPress/gutenberg-design @t-hamano in the context of the ongoing effort to standardize sizes of controls and buttons, see https://github.com/WordPress/gutenberg/issues/46734, can you please clarify what size these buttons should use?

I'm not sure this is a case that hae been taken into consideration on https://github.com/WordPress/gutenberg/issues/46734
These buttons are either visually shown _inside_ the input field or on the right of the input field.
Assuming the input field must now be 40 pixels, should the buttons be 40 pixels as well? Thank you.

Screenshot 2024-02-01 at 09 54 07

**Review by t-hamano — CHANGES_REQUESTED** [review]
As far as I know, `URLInput` component has been around for a very long time, as evidenced by the fact that they are class-based components. I think the fundamental problem is that we are using the `input` element itself, which requires various style customizations.

My guess is that this `input` element can be replaced with a `TextControl` component. If we do so, I think that way we can restore focus style to the default and at the same time remove many CSS styles (padding, margin, height, border etc).

In the future, as mentioned in #50893, it may be replaced with the `LinkControl` component.

> Assuming the input field must now be 40 pixels, should the buttons be 40 pixels as well?

In this PR, the button size is back to 36px, so I think we should change it to 32px or 40px.
How about changing the button size to 32px and aligning it vertically with the 40px input field?

Additionally, we will also need to adjust the balance of the layout when opening the link settings.

image

**Comment by afercia** [comment]
> My guess is that this input element can be replaced with a TextControl component. 

Sure, standardizing would be welcome. However, that would a bigger refactoring that is out of the scope of this PR. We should always try to keep changes limited to the specific issue we're trying to solve. This PR only aims to:
- Restore a correct focus style, which implies restoring the input field border.
- Adjust the CSS so that the restored border doesn't look ugly, limiting the changes to the bare minimum.
- Improve consistency and use the new controls and button sizes.

Anything else should go in a separate issue. I already created https://github.com/WordPress/gutenberg/issues/58507

> How about changing the button size to 32px and aligning it vertically with the 40px input field?

If we make the inpput field 40 and the buttons 32, it won't look that nice especially for the button _inside_ the input field. I'd rather make them 40. Screenshot with focused buttons to illustrate how it would look with 32:

Screenshot 2024-02-01 at 12 36 56

> we will also need to adjust the balance of the layout when opening the link settings.

Good point, thank you.

Edit: Ok I will use 32 for the buttons, as it's the simplest solution for now.

**Comment by afercia** [comment]
The `keyboardReturn` icon could be improved. The SVG glyph isn't really centered within the `viewBox` so that while the button is alignred, the icon doesn't look aligned. Zoomed-in screenshot with background colors applies to better illustrate:

keyboardReturn

I will create a separate issue.

**Comment by afercia** [comment]
Noting that the image link settings spacing, left/right padding and top border look broken on trunk as well. Screenshot from current trunk:

Screenshot 2024-02-01 at 14 24 54

I'll try to adjust that a little keeping changes to a minimum as that's out of the scope of this issue. Further adjustments should be handled in a separate issue, see https://github.com/WordPress/gutenberg/issues/58507

**Comment by afercia** [comment]
Screenshots after latest commits. All buttons focused to better illustrate their size:

Screenshot 2024-02-01 at 14 50 54

**Comment by annezazu** [comment]
@richtabor just in case you have any thoughts here after you've done so much work with link control in general.

File: packages/block-editor/src/components/link-control/README.md
Comment (t-hamano) [inline packages/block-editor/src/components/link-control/README.md]: Since the code example focuses on `LinkControlSearchInput`, I believe this prop on the `Button` component is unnecessary.

File: packages/block-editor/src/components/url-popover/style.scss
Comment (t-hamano) [inline packages/block-editor/src/components/url-popover/style.scss]: To me, this padding seems a bit large. How about just applying `padding: $grid-unit-10`?

File: packages/block-editor/src/components/url-input/style.scss
Comment (t-hamano) [inline packages/block-editor/src/components/url-input/style.scss]: Is there a reason why only the left padding is large? How about unifying it with `$grid-unit-10`?

File: packages/block-editor/src/components/url-popover/style.scss
Comment (afercia) [reply packages/block-editor/src/components/url-popover/style.scss]: This is the _container_ of the image URL input field. For clarity, the one with lightbox background in the screenshot below. For consistency with the normal link input field UI, the padding must be 16 pixels The only difference is that the image URL input has buttons outside of the input on the right so the right padding must be smaller for a better balance, and it's unchanged: it was 8 pixels also before.

Screenshot 2024-02-05 at 11 57 46

File: packages/block-editor/src/components/url-input/style.scss
Comment (afercia) [reply packages/block-editor/src/components/url-input/style.scss]: Yes there is a reason. For consistency with the normal link URL input field, where the left padding is 16 pixels. Pn trunk, besides the inconsistent focus style, the left padding is cleaerly inconsistent. See screenshot below.
Worth reminding there's some large inconsistency with the padding of form controls throughout the editor UI and that's tracked in https://github.com/WordPress/gutenberg/issues/57394 It would be great to give that issue some priority.

Screenshot 2024-02-05 at 12 12 45

File: packages/block-editor/src/components/link-control/README.md
Comment (afercia) [reply packages/block-editor/src/components/link-control/README.md]: I'm not sure I agree. Your argumentation could then be made for other things in this example, for example the `HStack` component would be unnecessary together with the `justify` prop. The `className` prop would be unnecessary. Instead, I think that code examples should be a full representation of the intended usage because developers will just copy and paste them. If the code example is incomplete or inaccurate, that would lead to inconsistencies or even bugs.

**Comment by richtabor** [comment]
> @richtabor just in case you have any thoughts here after you've done so much work with link control in general.

Instead of dedicating a lot of time to force this component to look and feel consistent with LinkControl, why not migrate it to actually use LinkControl, as detailed in https://github.com/WordPress/gutenberg/issues/50893? That way we get all the benefits of LinkControl, and a consistent linking experience wherever you are in WordPress.

**Comment by afercia** [comment]
> > @richtabor just in case you have any thoughts here after you've done so much work with link control in general.
> 
> Instead of dedicating a lot of time to force this component to look and feel consistent with LinkControl, why not migrate it to actually use LinkControl, as detailed in #50893? That way we get all the benefits of LinkControl, and a consistent linking experience wherever you are in WordPress.

@richtabor sure, that would be welcome. As I mentioned earlier, this PR is about accessibility and only aims to restore a correct focus style. It brings in some additional CSS changes for the sole purpose to not make these input fields look ugly and inconsistent.

Any other change or bigger refactoring should go in a separate issue.

Edit: Oh well I meant that _there is an issue,_  but no PR yet for that refactoring. In the meantime this is a smaller, self-contained improvement that can land in the next WordPress release, while I guess that for a larger refactoring we'll have to wait a little longer.

File: packages/block-editor/src/components/link-control/README.md
Comment (t-hamano) [reply packages/block-editor/src/components/link-control/README.md]: I see. I don't have a strong opinion on this, so either is fine. However, I was just curious because there was no code example of opting in this prop to the `Button` component in the Gutenberg documentation.

File: packages/block-editor/src/components/url-input/style.scss
Comment (t-hamano) [reply packages/block-editor/src/components/url-input/style.scss]: I see, so how about making the padding on the right side `$grid-unit-20` as well? The URL field of the MediaReplaceFlow component has 16px padding applied to the left and right sides.

image

Alternatively, to narrow the scope of this PR, it might be a good idea to revert the padding change and address it in the future with #57394.

Comment (afercia) [reply packages/block-editor/src/components/url-input/style.scss]: Fair enough. I will change the left padding. I'd be more glad to change it if I knew what the correct padding to use is. As I said, there's large inconsistency across the editor and that should be addressed in https://github.com/WordPress/gutenberg/issues/57394.
Personally, I don't like the 16 pixels _left_ padding. To better clarify: I don't like left padding when it's uneven with top and bottom padding. I think that's visually unbalanced and doesn't help readability. It feels like an unnecessary 'empty'  space. Regarding the right padding, I think it should be addressed in the context of a new issue I created, see https://github.com/WordPress/gutenberg/issues/58679

**Comment by afercia** [comment]
Latest commit reduces the left padding to 12 pixels. Screenshot of the two URL input UIs for comparison:

Screenshot 2024-02-06 at 12 50 59

**Comment by t-hamano** [comment]
Similar component changes were made in #58666. I think #58666 is ready to ship, so after #58666 is merged, it might be a good idea to merge the trunk into this PR and then adjust it.

**Comment by afercia** [comment]
I see that in the meantime this PR https://github.com/WordPress/gutenberg/pull/58666, which is not accompanied by a related issue to allow broader discussion, introduced some conflicting changes. I'm not sure I agree with some of those changes so I'll rebase this PR and try to address some new concerns.

The most important concern is the removal of a visible border for the input fields. See:

https://github.com/WordPress/gutenberg/blob/30b434956321d95f2d4acef83605b85f7546d61e/packages/block-editor/src/components/url-input/style.scss#L21-L23

Borderless input fields are a no-go for accessibility. 
Also, I don't think overriding the base components styling with local, ad-hoc, CSS is any good. Base components should be used throughout the editor unchanged. If there is a good case for a new styling, I'd suggest to:
- Create an issue to illustrate the use case for new styling or functionality.
- Propose a new variant for the base component.
- Implement the variant in the base compoent _and document it_.
- Use the variant instead of ad-hoc overrides.

Ad-hoc overrides defeat the purpose of having a library of reusable UI components in the first place.
In the long term, ad-hoc overrides introduce larce inconsistencies and technical debt, which is less than ideal.

Cc @ciampo @mirka as maintainers of the components package.

**Comment by afercia** [comment]
Looking more in depth into the changes from https://github.com/WordPress/gutenberg/pull/58666

Yes, the focus style is restored and it's consistent. Spacing and padding is inconsistent though:

Screenshot 2024-02-07 at 12 29 46

The default styling is borderless. This isn't okay for accessibility and, as mentioned earlier, it's an override of the base component that should not be allowed. 

Screenshot 2024-02-07 at 11 04 36

The border removal impacts also the LinkControl UI used for blocks other than the ones with images:

Screenshot 2024-02-07 at 09 47 16

Playing devil's advocate, as it would be an edge case: it is possible to enter a space in the input field and blur it so that it becomes an empty field with absolutely no visual shape or anything else that makes the input field visible. It just looks like an empty white space and it feels broken. Screenshot:

Screenshot 2024-02-07 at 11 04 05

Lastly, the link settings panel has the same layout problems that were reported on this issue during code review and then addressed, but they haven't been reported on https://github.com/WordPress/gutenberg/pull/58666 so that the layout on trunk is still problematic:

Screenshot 2024-02-07 at 09 43 21

**Comment by t-hamano** [comment]
Let me summarize the main points of this PR once again. I also agree with applying a default border. We should be able to do it just by deleting this: 

https://github.com/WordPress/gutenberg/blob/87c5697df786d168f585968c4b71c794f186b72b/packages/block-editor/src/components/url-input/style.scss#L21-L23

In addition, I think we can include the following improvements attempted in this PR if necessary.

- Change input field height to 40px
- Align the input field and button vertically
- Layout when opening link settings

I think adjusting the input field padding can be handled separately with #57394.

**Comment by afercia** [comment]
Rebased on top of latest trunk and restored the inputs default border. I updated this PR title accordingly.

Screenshots: default and focused:

Screenshot 2024-02-07 at 13 13 39

**Comment by t-hamano** [comment]
I restored the top border that was originally in the link settings by [a6ad295](https://github.com/WordPress/gutenberg/pull/58505/commits/a6ad2957623f816892225cf77668b3189e71e4bb). What do you think?

https://github.com/WordPress/gutenberg/assets/54422211/ff2a3722-41b3-464d-8e2e-43b82f87c3a7

File: packages/block-editor/src/components/url-popover/style.scss
Comment (t-hamano) [reply packages/block-editor/src/components/url-popover/style.scss]: This padding also takes effect after the URL input is _submitted_. To me, this padding seems a bit large. How about leaving the padding at its original value to narrow down the focus for this PR?

image

**Comment by afercia** [comment]
> I restored the top border that was originally in the link settings by [a6ad295](https://github.com/WordPress/gutenberg/pull/58505/commits/a6ad2957623f816892225cf77668b3189e71e4bb). What do you think?

I'd rather remove the border. Actually, I removed it intentionally for two reasons:
- For consistency: The LinkControl settings UI doesn't have any border, see screenshot below.
- It's the simplest change to solve the problem.

While waiting for https://github.com/WordPress/gutenberg/issues/50893, I think the closer these two UIs look, the better.

The LinkControl settings UI:

Screenshot 2024-02-07 at 13 56 46

**Comment by t-hamano** [comment]
> I'd rather remove the border. Actually, I removed it intentionally for two reasons:

Well, I guess that makes sense then. Please feel free to revert my changes 👍

P.S. The unit test is also failing on trunk, so this PR isn't the cause.

**Comment by afercia** [comment]
> P.S. The unit test is also failing on trunk, so this PR isn't the cause.

Oh I see, thanks @t-hamano and thanks for your always accurate reviews and feedback 🙇🏽

**Comment by richtabor** [comment]
> For consistency: The LinkControl settings UI doesn't have any border, see screenshot below.

For reference, the LinkControl isn't a relevant reference, as its extraneous control is being removed: https://github.com/WordPress/gutenberg/pull/58744

**Comment by richtabor** [comment]
@afercia can we try the treatment of the "Apply" icon to resembler LinkControl? As-is the popover is quite wide, covering a large portion of the image below.  

We need to also reduce the doubled padding in this view: 

CleanShot 2024-02-07 at 09 25 44

We may be able to use the settings icon (instead of the down caret), with an active state, to communicate the state and function better: 

CleanShot 2024-02-07 at 09 26 56

**Comment by richtabor** [comment]
Once we roll this is we really should look at bringing the full LinkControl proper to the image and media & text blocks. That way we don't have these disjointed experiences that feel a bit hacked together.

**Comment by t-hamano** [comment]
#58666 reported an issue where the text link edit popover border disappeared ([See](https://github.com/WordPress/gutenberg/pull/58666#discussion_r1481837919)). Once this PR is merged, that issue will also be fixed. For this PR, I think it's better to avoid new attempts as much as possible and focus on fixing existing problems.

**Comment by afercia** [comment]
> #58666 reported an issue where the text link edit popover border disappeared ([See](https://github.com/WordPress/gutenberg/pull/58666#discussion_r1481837919)). 

Yes, and it is also possible to just Tab from the input field to the Submit button. In this case, the borderless input field looks just wrong. As I mentioned in that PR, I'm not sure overriding the base component styling should ever be allowed in the first place. Also, borderless inputs aren't ok for accessibility and seems to be not a great idea in general.

Also, it appears the changes in https://github.com/WordPress/gutenberg/pull/58666 haven't been sufficiently tested and clearly not tested when using a keyboard, which is less than ideal.

Screenshot:

Screenshot 2024-02-08 at 10 54 06

**Comment by afercia** [comment]
> @afercia can we try the treatment of the "Apply" icon to resembler LinkControl? As-is the popover is quite wide, covering a large portion of the image below.

@richtabor I don't like the visual placement of buttons inside input fields and I'm not sure it's any good for accessibility and usability, at the point I would like that to be reconsidered in the LinkControl component.

It's a good practice to keep patches and PRs in the scope of the specific issue to solve and keep them self-contained. The purpose of this PR is to restore the border and the focus style. It's already doing too much and I'm not sure adding more changes is good. I'm glad to collect and report the remaining points in a saparate issue. 

> I think it's better to avoid new attempts as much as possible 

@t-hamano agreed thanks. When you have a chance, I'd appreciate a final review and approval as I can't merge this PR without it being approved.

File: packages/block-editor/src/components/url-popover/style.scss
Comment (afercia) [reply packages/block-editor/src/components/url-popover/style.scss]: To me, the padding looks okay and it's more consistent with the enw UI of the LinkControl which is not surprising because that padding is supposed to be 16 pixels, accordingly to teh mockups and design I've seen.

Screenshots: before, after, and compared to the LinkControl where you can see the spacing is almost identical:

Screenshot 2024-02-08 at 15 04 06

I'm not sure this PR is the right place to over-optimize the layout. This PR has a different scope. Moreover, the UI will be sooner or later replaced with the LinkControl. I'm not against further improvements but if you feel there's room for (unrelated) layout adjustments I'd kindly ask you to please create a separate issue.

**Comment by richtabor** [comment]
> @richtabor I don't like the visual placement of buttons inside input fields and I'm not sure it's any good for accessibility and usability, at the point I would like that to be reconsidered in the LinkControl component.

We shouldn't be leaning on personal preference. This is a common, expected, pattern across a myriad of software applications and experiences. 

> It's a good practice to keep patches and PRs in the scope of the specific issue to solve and keep them self-contained. The purpose of this PR is to restore the border and the focus style.

This PR is touching the metrics more than anything (https://github.com/WordPress/gutenberg/pull/58505/files). 

I had an original, much smaller PR (https://github.com/WordPress/gutenberg/pull/58780) to target a fix (though your solution on that particular part is better). Should we consider pulling that into its own fix?

From my POV, I'm not partial to pushing indirect changes without holistic solutions. If we merge as-is, with the UI changes (apart from the border style fix), we will be introducing a less-than-ideal experience for one of the most used blocks in the editor.

It's much more challenging to get approval when a solution isn't complete.

**Comment by afercia** [comment]
I reported the main pending points in a separate issue at https://github.com/WordPress/gutenberg/issues/58840

**Comment by t-hamano** [comment]
My biggest concern is that the final release of Gutenberg, which will be bundled with WP6.5, is just around the corner. Once that version is released, _enhancements_ should not be backportable. We need more time to find the ideal solution. I think we need to find a realistic landing point for WP6.5.

Once again, I would like to mention the problems with the current trunk that I think should at least be fixed in this PR.


Input field in the inline text URL popover has no default border (Regression with #58666):

image

Input field in URL popover opened from block toolbar has no default border:

image

Border layout is incorrect when opening the Link Settings

image

@afercia @richtabor What is your best solution in WP6.5?

**Comment by richtabor** [comment]
> Input field in the inline text URL popover has no default border (Regression with https://github.com/WordPress/gutenberg/pull/58666):

I say we fix this, and.. 

> Input field in URL popover opened from block toolbar has no default border:

add the border here (if not fixed by the above). 

If they're separate, that's fine.

**Comment by t-hamano** [comment]
I appreciate the work @afercia has done on this PR. But how about following the title of this PR and simply restoring the default border? All we need to do is make the following changes in trunk:

```diff
diff --git a/packages/block-editor/src/components/url-input/style.scss b/packages/block-editor/src/components/url-input/style.scss
index 66f71e8035..f8d84ae43b 100644
--- a/packages/block-editor/src/components/url-input/style.scss
+++ b/packages/block-editor/src/components/url-input/style.scss
@@ -18,10 +18,6 @@ $input-size: 300px;
                margin-left: 0;
                margin-right: 0;
 
-               &:not(:focus) {
-                       border-color: transparent;
-               }
-
                /* Fonts smaller than 16px causes mobile safari to zoom. */
                font-size: $mobile-text-min-font-size;
                @include break-small {
```

The broken border layout when opening Link Setting is a bug, so it should be possible to backport it even after the release of WP6.5 Beta1 (Of course, I also think it's a bug that the input field doesn't have a default border).

We should be able to refer to the discussion in this PR when improving the layout and size in the future.

**Comment by afercia** [comment]
> enhancements should not be backportable.

@t-hamano I'm not sure I understand your point. The lack of consistent focus style (already fixed elsewhere) and the lack of the default border are bugs, not enhancements.

This PR originally contained only the very minimum of CSS changes necessary to make the restored focus style and border not look ugly. 

It now contains much more than originally needed because I received feedback during code review to 'how about fixing also this' and 'how about adjusting also that'... I tried to address the received feedback but I see now that maybe I shouldn't have done that. All those unnecessary adjustments made this PR take much longer than needed. I would like to point out that I remarked a few times to please try to keep this PR focused on its original scope. It is now bigger than planned because of the feedback I received, including the feedback I received from you.

Overall, I think this PR fixes a bug and improves the layout. Changes don't need to be perfect. They need to be good enough.

Lastly, I'm not sure the overall process followed in this PR is ideal to make contributors work pleasant and efficient. I know everybody's intent is good, but we actually wasted a lot of time on this PR.

Fair enough, I will revert all the changes except the one that restores the border. It will look ugly. 

For the future, I'd kindly ask everyone to be more respectful of other contributors time.

**Comment by t-hamano** [comment]
> For the future, I'd kindly ask everyone to be more respectful of other contributors time.

I apologize if my reviews have been inconsistent. However, like all contributors, I have limited time to contribute. I will try to do my best in all reviews within that time.

**Comment by afercia** [comment]
In the latest commit I reverted all changes except the restored border.

See attached screenshot below with the various states and comparison with the LinkControl UI.

All the small glitches and inconsistencies are now back, including:

- input height 
- input padding 
- buttons vertical alignment 
- top border of the settings panel 
- padding of the link preview

I would like to note that if this is what the prevalent opinion on this PR really wants, then I'm afraid we have different ideas on what is good to be merged, and what is best for a better user experience.

Screenshot 2024-02-09 at 15 58 59

**Review by t-hamano — APPROVED** [review]
LGTM!

> I would like to note that if this is what the prevalent opinion on this PR really wants, then I'm afraid we have different ideas on what is good to be merged, and what is best for a better user experience.

I understand this, of course. After merging this PR, we will still need to continue the discussion.

**Comment by afercia** [comment]
> I understand this, of course. After merging this PR, we will still need to continue the discussion.

Sure thanks. For clarity, I'm not against accurate revisions. I would love to see the same level of accuracy for _all_ the pull requests, including the ones that broke the border and focus style.

**Comment by t-hamano** [comment]
This PR is not part of the final Gutenberg version 17.7 that will be bundled with WP6.5, so it must be backported.

This PR fixes the "bug" of missing default focus, and also fixes the "regression" that occurred in #58666, so it should be backportable.

**Comment by youknowriad** [comment]
I just cherry-picked this PR to the backports/beta1 branch to get it included in the next release: f8897cc92b

**Pull request merged**
